File: obfuscator/obfuscationFuns/addRandomValue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function addRandomValue(input, keyArray, initor) {
    const obfuscated = new Uint8Array(input.length);
    for (let i = 0; i < input.length; i++) {
        obfuscated[i] = (input[i] + initor) % 256; // Addition with modulus
    }
    return obfuscated;
}
function de_addRandomValue(input, keyArray, initor) {
    const deobfuscated = new Uint8Array(input.length);
    for (let i = 0; i < input.length; i++) {
        deobfuscated[i] = (input[i] - initor + 256) % 256; // Subtraction with modulus
    }
    return deobfuscated;
}
function generateRandomValue() {
    return Math.floor(Math.random() * 256);
}
let funPair = {
    obfuscation: addRandomValue,
    deobfuscation: de_addRandomValue,
    initorFn: generateRandomValue
};
exports.default = funPair;

File: android/wireguard_obfuscation/obfuscationFuns/AddRandomValue.java
package com.morph_vpn.plugin.wireguard_obfuscation.obfuscationFuns;
import java.util.Arrays;
import java.util.Random;

public class AddRandomValue {
    public static byte[] addRandomValue(byte[] input, byte[] keyArray, Object initor) {
        byte[] obfuscated = new byte[input.length];

        for (int i = 0; i < input.length; i++) {
            obfuscated[i] = (byte) ((input[i] + ((Integer)initor).intValue()) % 256); // Addition with modulus
        }

        return obfuscated;
    }

    public static byte[] de_addRandomValue(byte[] input, byte[] keyArray, Object initor) {
        byte[] deobfuscated = new byte[input.length];

        for (int i = 0; i < input.length; i++) {
            deobfuscated[i] = (byte) ((input[i] - ((Integer)initor).intValue() + 256) % 256); // Subtraction with modulus
        }

        return deobfuscated;
    }

    public static int generateRandomValue() {
        Random random = new Random();
        return random.nextInt(256);
    }

    public static void main(String[] args) {
        byte[] input = new byte[]{/* initialize input byte array */};
        byte[] keyArray = new byte[]{/* initialize keyArray byte array */};

        Object initor = generateRandomValue();

        byte[] obfuscated = addRandomValue(input, keyArray, initor);
        byte[] deobfuscated = de_addRandomValue(obfuscated, keyArray, initor);

        System.out.println("Input: " + Arrays.toString(input));
        System.out.println("Obfuscated: " + Arrays.toString(obfuscated));
        System.out.println("Deobfuscated: " + Arrays.toString(deobfuscated));
    }
}
